src/Orchard.Web/Modules/Orchard.OpenId/Module.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Orchard.Environment.Shell;
+using Orchard.Security;
+using Orchard.OpenId.Indexes;
+using Orchard.OpenId.Models;
+using Orchard.Users.Models;
+using Orchard.OpenId.Services;
+using Orchard.FileSystem.AppData;
+using Microsoft.Extensions.Logging;
+using Orchard.Settings;
+
+namespace Orchard.OpenId
+{
+    public class Startup : StartupBase
+    {     
+        private readonly string _certificateFullPath;
+        private const string certificateFileName = "Certificate.pfx";
+        private readonly string tenant;
+
+        public Startup(ShellSettings shellSettings, IAppDataFolder appDataFolder, ILoggerFactory loggerFactory)   
+        {
+            tenant = shellSettings.Name;
+            _certificateFullPath = appDataFolder.Combine(shellSettings.Name, certificateFileName);
+        }
+
+        public override void Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var baseUrl = serviceProvider.GetRequiredService<ISiteService>().GetSiteSettingsAsync().Result.BaseUrl ?? tenant;
+            builder.UseOpenIddict();
```

Comment:
Also, just to be sure you understand how multi-tenancy works in this case, each tenant will have their own middleware, meaning that this `UseOpenIddict` call might run for as many tenants there are in the site. It's options should reflect it by getting a tenant name somehow, so that JWT can't be shared accross different tenants. You can see an example in the USers module when the authentication cookie is configured, and its path is limited to the tenant name in the url.